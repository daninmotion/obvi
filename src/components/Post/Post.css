.title {
	font-size: 30px;
	color: #283C46;
}

.link {
	color: #283C46;
}

.link:focus,
.link:hover {
	color: #000000;
	transition: color 0.3s ease;
}

.content {
	font-size: 16px;
	line-height: 1.8;
}