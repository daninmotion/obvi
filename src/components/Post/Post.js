import React from 'react'
import styles from './Post.css'

const Post = () => (
	<div>
		<h2><a href='#'>Blog title</a></h2>
		<p>Content</p>
	</div>
)

export default Post