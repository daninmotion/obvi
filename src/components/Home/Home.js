import React from 'react'
import Post from '../Post/Post'

const Home = ({posts}) => (
	<div>
		{posts.length
		? posts.map(post => <Post {...post} key={post.slug} titleLink={true} />) : <p>Awating posts!</p>}
	</div>
)

export default Home