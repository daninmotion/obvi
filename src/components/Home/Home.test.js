import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'
import Post from '../Post/Post'

test('Home component should render as expected', () => {
	const posts = [{
		"title": "How to build a progressive web app with React",
    "slug": "how-to-build-a-progressive-web-app-with-react",
    "excerpt": "One of the easiest...",
    "content": "One of the easiest ways to get a job in the early GitHub days was to blah."
	}]
	const component = shallow(<Home posts={posts} />)
	const tree = toJson(component)
	expect(component.find(Post).length).toBe(1)
	expect(tree).toMatchSnapshot()
})

test('Home component should render as expected', () => {
	const posts = [{
		"title": "How to build a progressive web app with React",
    "slug": "how-to-build-a-progressive-web-app-with-react",
    "excerpt": "One of the easiest...",
    "content": "One of the easiest ways to get a job in the early GitHub days was to blah."
	},
	{
		"title": "How to build a progressive web app with React",
    "slug": "how-to-build-a-progressive-web-app-with-react",
    "excerpt": "One of the easiest...",
    "content": "One of the easiest ways to get a job in the early GitHub days was to blah."
	}]
	const component = shallow(<Home posts={posts} />)
	const tree = toJson(component)
	expect(component.find(Post).length).toBe(2)
	expect(tree).toMatchSnapshot()
})

test('Home component should render as expected', () => {
	const component = shallow(<Home posts={[]} />)
	const tree = toJson(component)
	expect(component.find(Post).length).toBe(0)
	expect(tree).toMatchSnapshot()
})



















