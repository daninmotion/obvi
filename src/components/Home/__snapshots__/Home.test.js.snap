// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render as expected 1`] = `
<div>
  <Post
    content="One of the easiest ways to get a job in the early GitHub days was to blah."
    excerpt="One of the easiest..."
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true}
  />
</div>
`;

exports[`Home component should render as expected 2`] = `
<div>
  <Post
    content="One of the easiest ways to get a job in the early GitHub days was to blah."
    excerpt="One of the easiest..."
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true}
  />
  <Post
    content="One of the easiest ways to get a job in the early GitHub days was to blah."
    excerpt="One of the easiest..."
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true}
  />
</div>
`;

exports[`Home component should render as expected 3`] = `
<div>
  <p>
    Awating posts!
  </p>
</div>
`;
